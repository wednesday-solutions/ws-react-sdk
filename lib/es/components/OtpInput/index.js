import{_ as t,a as e,d as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"../_rollup-plugin-inject-process-env-1259e43d.js";import a from"react";import i from"styled-components";import n from"prop-types";import{translate as o}from"../IntlGlobalProvider/index.js";import s from"../Timer/index.js";import{c as l}from"../colors-93a064d8.js";import{u as p,r as m}from"../cognito-cd75d0f5.js";import c from"../CustomInput/index.js";import"antd/lib/message/style";import d from"antd/lib/message";import"antd/lib/select/style";import"antd/lib/select";import"react-intl";import"../T/index.js";import"../If/index.js";import"../fonts-ca6864a0.js";import"../media-ee9cbc51.js";import"amazon-cognito-identity-js";import"antd/lib/input/style";import"antd/lib/input";var f,g,h,u="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' viewBox='0 0 20 20'%3e %3cg%3e %3cpath fill='%23af0974' d='M10 0A10 10 0 1 1 0 10 10 10 0 0 1 10 0z' transform='translate(-156 -79) translate(156 79)'/%3e %3cpath fill='white' d='M13 5.806L12.194 5 9 8.194 5.806 5 5 5.806 8.194 9 5 12.194l.806.806L9 9.806 12.194 13l.806-.806L9.806 9z' transform='translate(-156 -79) translate(157 80)'/%3e %3c/g%3e%3c/svg%3e",w="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' viewBox='0 0 20 20'%3e %3cg%3e %3cpath fill='%232fbb8a' d='M10 0A10 10 0 1 1 0 10 10 10 0 0 1 10 0z' transform='translate(-156 -79) translate(156 79)'/%3e %3cpath fill='white' d='M6.582 11.623L4.2 9.236l-.8.8 3.182 3.182L13.4 6.4l-.8-.8z' transform='translate(-156 -79) translate(158 80)'/%3e %3c/g%3e%3c/svg%3e",v="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='22' height='20' viewBox='0 0 22 20'%3e %3cg%3e %3cpath fill='%23ffbc41' d='M9.248 3.186a2 2 0 0 1 3.5 0l7.617 13.85A2 2 0 0 1 18.617 20H3.383a2 2 0 0 1-1.753-2.964z' transform='translate(-154 -78) translate(154 78)'/%3e %3cpath fill='white' d='M.846 1.541h1.417l.192-5.8H.642zm1.766 1.825a.983.983 0 0 0-1.021-.985.986.986 0 1 0 0 1.97.983.983 0 0 0 1.021-.985z' transform='translate(-154 -78) translate(163.648 90.371)'/%3e %3c/g%3e%3c/svg%3e",b=1,x=2,y=3,j=4,z=i.img(f||(f=t(["\n  margin-right: 10px;\n"]))),_=function(t){var e=arguments.length>1&&void 0!==arguments[1]?arguments[1]:j;if(t){var r,i;switch(e){case b:r=d.success,i=w;break;case x:r=d.error,i=u;break;case y:r=d.warning,i=v;break;default:r=d.info,i=v}r({content:t,icon:a.createElement(z,{src:i})})}},M=i.span(g||(g=t(["\n  background-color: ",";\n  background: ",";\n"])),l.white,l.white),k=i(c)(h||(h=t(["\n  &.ant-input-affix-wrapper {\n    border-radius: 8px;\n    border: solid 1px ",";\n  }\n  &.ant-input-affix-wrapper > input.ant-input {\n    border: none;\n    max-height: 3rem;\n  }\n"])),l.secondary);function L(){return(L=r((function*(t){try{t.verifyNumber?yield p(t.email,t.phone):yield m(t.email),_(o("otp_resent"),b)}catch(t){var e=t.message||o("failed_to_send_otp");throw _(e||t,x),t}}))).apply(this,arguments)}var B=t=>a.createElement(M,null,a.createElement(s,{initialMinute:0,initialSeconds:59,retryNo:5,resendCallback:()=>function(t){return L.apply(this,arguments)}(t)}));function E(t){return a.createElement(k,e({"data-testid":"otp-input"},t,{suffix:B(t),placeholder:o("otp_msg")}))}E.propTypes={verifyNumber:n.bool,phone:n.string};export default E;
