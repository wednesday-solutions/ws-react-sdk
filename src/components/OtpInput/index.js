/**
 *
 * OtpInput
 *
 */

import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { translate } from '@components/IntlGlobalProvider';
import Timer from '@components/Timer';
import { colors } from '@themes';
import { resendConfirmationCode, updatePhoneNumberRequest } from '@services/cognito';
import CustomInput from '@components/CustomInput';
import { NOTIFICATION_TYPE, showNotification } from '@utils';

const ResendWrapper = styled.span`
  background-color: ${colors.white};
  background: ${colors.white};
`;
const OTP = styled(CustomInput)`
  &.ant-input-affix-wrapper {
    border-radius: 8px;
    border: solid 1px ${colors.secondary};
  }
  &.ant-input-affix-wrapper > input.ant-input {
    border: none;
    max-height: 3rem;
  }
`;

async function resendOTP(props) {
  try {
    if (props.verifyNumber) {
      await updatePhoneNumberRequest(props.email, props.phone);
    } else {
      await resendConfirmationCode(props.email);
    }
    showNotification(translate('otp_resent'), NOTIFICATION_TYPE.SUCCESS);
  } catch (e) {
    const message = e.message || translate('failed_to_send_otp');
    showNotification(message || e, NOTIFICATION_TYPE.ERROR);
    throw e;
  }
}

const renderResendOTP = props => (
  <ResendWrapper>
    <Timer initialMinute={0} initialSeconds={0} retryNo={5} resendCallback={() => resendOTP(props)} />
  </ResendWrapper>
);

function OtpInput(props) {
  return (
    <OTP data-testid={'otp-input'} {...props} suffix={renderResendOTP(props)} placeholder={translate('otp_msg')} />
  );
}

OtpInput.propTypes = {
  verifyNumber: PropTypes.bool,
  phone: PropTypes.string
};

export default OtpInput;
