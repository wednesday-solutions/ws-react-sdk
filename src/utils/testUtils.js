import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
