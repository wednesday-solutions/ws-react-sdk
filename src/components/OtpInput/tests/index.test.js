/**
 *
 * Tests for OtpInput
 *
 */

import React from 'react';
import { renderWithIntl, timeout } from '@utils/testUtils';
import OtpInput from '../index';
import { fireEvent } from '@testing-library/dom';

describe('<OtpInput />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<OtpInput />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 OtpInput component', () => {
    const { getAllByTestId } = renderWithIntl(<OtpInput />);
    expect(getAllByTestId('otp-input').length).toBe(1);
  });
  it('should ensure that if no is verified it should call updatePhoneNumberRequest  ', () => {
    const number = '9991232319';
    const { getByTestId } = renderWithIntl(<OtpInput verifyNumber={true} phone={number} />);
    fireEvent.click(getByTestId('timer'));
    timeout(300);
  });
});
