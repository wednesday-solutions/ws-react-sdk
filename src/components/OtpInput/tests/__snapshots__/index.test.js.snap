// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtpInput /> should render and match the snapshot 1`] = `
<body>
  <div>
    <span
      class="ant-input-affix-wrapper sc-eCApnc cZuBYE sc-gKAaRy hJsanL"
    >
      <input
        class="ant-input"
        data-testid="otp-input"
        placeholder="otp_msg"
        type="text"
        value=""
      />
      <span
        class="ant-input-suffix"
      >
        <span
          class="sc-jSFjdj efObN"
        >
          <div
            data-testid="timer"
          >
            <p
              class="sc-bdnxRM iIGKAt sc-gtsrHT jSvlHI"
              data-testid="t"
            >
              RESEND
            </p>
          </div>
        </span>
      </span>
    </span>
  </div>
</body>
`;
